common/src/test/java/io/netty/util/HashedWheelTimerTest.java

```diff
@@ -185,54 +179,53 @@ public void testRejectedExecutionExceptionWhenTooManyTimeoutsAreAddedBackToBack(
     @Test
     public void testNewTimeoutShouldStopThrowingRejectedExecutionExceptionWhenExistingTimeoutIsCancelled()
         throws InterruptedException {
-        final HashedWheelTimer timer = new HashedWheelTimer(Executors.defaultThreadFactory(), 100,
+        final int tickDurationMs = 100;
+        final HashedWheelTimer timer = new HashedWheelTimer(Executors.defaultThreadFactory(), tickDurationMs,
             TimeUnit.MILLISECONDS, 32, true, 2);
-        timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
-        Timeout timeoutToCancel = timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
-        timeoutToCancel.cancel();
-        waitUntilNewTimeoutCanBeAdded(200, timer, 1000);
+        timer.newTimeout(createNoOpTimerTask(), 5, TimeUnit.SECONDS);
+        Timeout timeoutToCancel = timer.newTimeout(createNoOpTimerTask(), 5, TimeUnit.SECONDS);
+        assertTrue(timeoutToCancel.cancel());
+
+        Thread.sleep(tickDurationMs * 5);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)